### PR TITLE
Integrate Vagrant

### PR DESCRIPTION
**Pull request by AlanHohn**
Automatic setup of a CodeCombat development environment using Vagrant, VirtualBox, and Ubuntu.

Once integrated, I can update the wiki to reflect the new instructions:
1. `vagrant up` from the root directory
2. `scripts/vagrant/brunch.sh` from the root directory
3. `scripts/vagrant/dev-server.sh` from the root directory

Once Brunch indicates that it's done compiling, visit http://localhost:3000.

The Vagrantfile and provisioning scripts were originally developed by dpen2000 and are hosted on GitHub under an MIT license. I am contributing based on the permissions under the MIT license to use, copy, modify, merge, publish, distribute, and sublicense without limitation.

### CODE REVIEW

**Comment by nwinter** [comment]
Awesome, thanks @AlanHohn! Can you also extend it to running `mongod` (maybe with the `--dpbath` argument if needed) so that the dev database is also running, like the dev server and brunch?

@popey456963 @ssfinney want to test it out?

**Comment by ssfinney** [comment]
Sure, give me a day or so

**Comment by popey456963** [comment]
I'll try setting it up now.

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+# Original content copyright (c) 2014 dpen2000 licensed under the MIT license
+
+VAGRANTFILE_API_VERSION = "2"
+
+Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
+  config.vm.box = "hashicorp/precise64"
```
Comment (popey456963) [inline Vagrantfile]: Should we use trusty rather than precise for our Ubuntu version?

**Comment by AlanHohn** [comment]
* Mongo: I think Ubuntu kicks it off automatically on install. When I `vagrant ssh` into the VM and type `mongo` it's running, and the node server seems to find it OK.

* Trusty: Absolutely. I'll do that next. I'd also like to make it possible to control the RAM & number of CPUs for the VM from environment variables, so people with better dev machines can devote more power to running it. (Right now default is 2048 MB / 2 CPU.)

I'm still learning the typical development flow, but it seems like people need to download the latest MongoDB dump from the production server pretty frequently. When they do that, are there any extra steps to save off any local user progress that I can automate?

**Comment by popey456963** [comment]
Being a completely naive user here, who has never used vagrant - 
 - Downloaded Vagrant (Windows)
 - Already have Virtual Box installed
 - Ran installer for Vagrant
 - Now have a vagrant command I can use in cmd.exe
Do I have to download the github repo, move to it and then run `vagrant up`?

**Comment by popey456963** [comment]
Have another, better image, showing why it breaks.

**Comment by popey456963** [comment]
Things that I have learnt:
 - Use Virtual Box 4.3.12, newer versions do not work.
 - Run `Vagrant Up` once to download trusty, run it again to actually start the server up.
 - Turn vb.gui on (made a pull request about this) while debugging.  Don't turn it on when everything works (probably gonna remove pull request).

**Comment by AlanHohn** [comment]
Thanks for the feedback.

I tried it using 4.3.22, but on a Mac. That may cause some of the differences. After I make the switch to trusty I will try it out on Mac and Windows. A single `vagrant up` should do.

**Comment by popey456963** [comment]
Interesting, it might be windows being buggy, it generally is with this sort of thing.

**Pull request merged**